### PR TITLE
refactor: include router directly in fresh

### PR DESCRIPTION
**Pull request by crowlKats**
This PR changes from depending on https://deno.land/x/rutt to instead include a reduced & simplified version of it.

### CODE REVIEW

File: src/server/context.ts [hunk 1]
```diff
@@ -521,16 +527,23 @@ export class ServerContext {
...
+              ...ctx,
+              params,
+              render: createRender(req, params),
+              renderNotFound: createUnknownRender(req, {}),
+            }),
+        };
       } else {
+        routes[route.pattern] = {};
         for (const [method, handler] of Object.entries(route.handler)) {
-          routes[`${method}@${route.pattern}`] = (req, ctx, params) =>
```
Comment (bartlomieju) [inline src/server/context.ts:533]: Is this a public API? Why change the name here?

File: src/server/context.ts [hunk 2]
```diff
@@ -442,12 +446,14 @@ export class ServerContext {
       const { localUrl, path, size, contentType, etag } of this.#staticFiles
     ) {
       const route = sanitizePathToRegex(path);
-      routes[`GET@${route}`] = this.#staticFileHandler(
```
Comment (bartlomieju) [inline src/server/context.ts:445]: Similarly here, why chance it?

**Review by lucacasonato — COMMENTED** [review]
This doesn't yet touch the middleware handling itself, right?

File: src/server/context.ts (hunk 1 above)
Comment (lucacasonato) [reply src/server/context.ts:533]: No, not a public API.

File: src/server/context.ts (hunk 2 above)
Comment (lucacasonato) [reply src/server/context.ts:445]: I assume Leo changed these to simplify the router (as we don't have to parse `xx@xx` syntax anymore).

**Comment by bartlomieju** [comment]
> This doesn't yet touch the middleware handling itself, right?

Correct, just upstreaming the router for now. Middleware will be done in a followup.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**